src/fasta-recs.py

```diff
@@ -11,8 +10,22 @@ def main():
     )
     args = argparser.parse_args()
 
-    print(f"Now I need to process the records in {args.fasta}")
+    # print(f"Now I need to process the records in {args.fasta}")
+
+    printer = ''
+    for line in (args.fasta):
+        if '>' in line:
+            if printer != '':
+                print(printer)
+                printer = ''
+            name = line.replace('>',  '')
+            name = name.strip()
+            printer += name + '\t'
```

Comment:
Concatenating with `+=` does work the way you want most of the time. Python now uses the optimisation it also has for lists when you append to strings in this way. But I don't believe that this is guaranteed to remain that way or that it is only an implementation detail that can change in the future. A safe solution that always works is to collect the pieces of the string in a list and then `''.join(list)` when you need the bits concatenated.

That being said, this solution is fine and works as linear time concatenation the way CPython is implemented at the moment.